packages/tracing-processing/src/services/db/dbService.ts

```diff
@@ -1,17 +1,189 @@
+                name: string;
+                origin: string;
+                external_id: string;
+              }>(
+                `SELECT name, origin, external_id FROM tracing.tenants WHERE id = $1`,
+                [tracing.tenantId],
+              );
+
+              if (!consumer) {
+                return enrichPurpose(
+                  record,
+                  tracing,
+                  eService,
+                  fullPurpose,
+                  null,
+                  null,
+                  "CONSUMER_NOT_FOUND",
+                );
+              }
+
+              if (!eService) {
+                return enrichPurpose(
+                  record,
+                  tracing,
+                  null,
+                  fullPurpose,
+                  consumer,
+                  null,
+                  "ESERVICE_NOT_FOUND",
+                );
+              }
+
+              const tenantEservice = await db.oneOrNone<EserviceSchema>(
+                `SELECT * FROM tracing.eservices WHERE (producer_id = $1 OR consumer_id = $1)`,
+                [tracing.tenantId],
+              );
+
+              if (!tenantEservice) {
+                return enrichPurpose(
+                  record,
+                  tracing,
+                  eService,
+                  fullPurpose,
+                  consumer,
+                  null,
+                  "ESERVICE_NOT_ASSOCIATED",
+                );
+              }
+
+              const producer = await db.oneOrNone<{
+                name: string;
+                origin: string;
+                external_id: string;
+              }>(
+                `SELECT name, origin, external_id FROM tracing.tenants WHERE id = $1`,
+                [eService.producer_id],
+              );
+
+              if (!producer) {
+                return enrichPurpose(
+                  record,
+                  tracing,
+                  eService,
+                  fullPurpose,
+                  consumer,
+                  null,
+                  "PRODUCER_NOT_FOUND",
+                );
+              }
+
+              return enrichPurpose(
+                record,
+                tracing,
+                eService,
+                fullPurpose,
+                consumer,
+                producer,
+                undefined,
+              );
+            } catch (error) {
+              throw getEnrichedPurposeError(
+                `Error fetching record for tracingId: ${tracing.tracingId}, purpose_id: ${record.purpose_id}: Details: ${error}`,
+              );
+            }
+          },
+        );
+
+        const enrichedPurposes = await Promise.all(fullRecordPromises);
+
+        return enrichedPurposes;
       } catch (error) {
-        throw genericInternalError(`Error getPurposesByTracingId: ${error}`);
+        throw genericInternalError(
+          `Error in getEnrichedPurpose function: ${error}`,
+        );
       }
     },
   };
 }
 
+function enrichPurpose(
```

Comment:
I suggest to split this function, one for successful enrichment and one for error message, and use stricter parameters.
For example, we cannot allow null values on successful response